src/main/java/eu/clarin/cmdi/rasa/linkResources/impl/ACDHCheckedLinkResource.java

```diff
@@ -138,77 +140,93 @@ private PreparedStatement getPreparedStatement(String defaultQuery, Optional<Che
+            try (PreparedStatement statement = getPreparedStatement(defaultQuery, filterOrNoop, queryInClause, addUrlParms)) {
 
-        try (PreparedStatement statement = getPreparedStatement(defaultQuery, filter, sb.toString())) {
-            try (ResultSet rs = statement.executeQuery()) {
-                try (Stream<Record> recordStream = DSL.using(con).fetchStream(rs)) {
-                    return recordStream.map(CheckedLink::new).collect(Collectors.toMap(CheckedLink::getUrl, Function.identity()));
+                try (ResultSet rs = statement.executeQuery()) {
+                    try (Stream<Record> recordStream = DSL.using(con).fetchStream(rs)) {
+                        return recordStream.map(CheckedLink::new).collect(Collectors.toMap(CheckedLink::getUrl, Function.identity()));
+                    }
                 }
             }
         }
 
     }
 
-
     @Override
     public Boolean save(CheckedLink checkedLink) throws SQLException {
 
         //get old checked link
-        CheckedLink oldCheckedLink = get(checkedLink.getUrl());
+        final Optional<CheckedLink> oldCheckedLink = get(checkedLink.getUrl());
 
-        if (oldCheckedLink != null) {
+        if (oldCheckedLink.isPresent()) {
             //save to history
-            saveToHistory(oldCheckedLink);
+            saveToHistory(oldCheckedLink.get());
 
             //delete it
             delete(checkedLink.getUrl());
         }
 
         //save new one
-        return insertCheckedLink(checkedLink, "status");
+        return insertCheckedLink(checkedLink, Table.STATUS);
     }
 
-    private PreparedStatement getInsertPreparedStatement(String tableName) throws SQLException {
-        PreparedStatement preparedStatement;
-        if (tableName.equals("status")) {
-            String insertStatusQuery = "INSERT INTO status(url,statusCode,method,contentType,byteSize,duration,timestamp,redirectCount,collection,record,expectedMimeType,message) VALUES (?,?,?,?,?,?,?,?,?,?,?,?)";
-            preparedStatement = con.prepareStatement(insertStatusQuery);
-        } else {//history
-            String insertHistoryQuery = "INSERT INTO history(url,statusCode,method,contentType,byteSize,duration,timestamp,redirectCount,collection,record,expectedMimeType,message) VALUES (?,?,?,?,?,?,?,?,?,?,?,?)";
-            preparedStatement = con.prepareStatement(insertHistoryQuery);
+    private PreparedStatement getInsertPreparedStatement(Table tableName) throws SQLException {
+        final String insertStatusQuery = "INSERT INTO status(url,statusCode,method,contentType,byteSize,duration,timestamp,redirectCount,collection,record,expectedMimeType,message) VALUES (?,?,?,?,?,?,?,?,?,?,?,?)";
+        final String insertHistoryQuery = "INSERT INTO history(url,statusCode,method,contentType,byteSize,duration,timestamp,redirectCount,collection,record,expectedMimeType,message) VALUES (?,?,?,?,?,?,?,?,?,?,?,?)";
+        switch (tableName) {
```

Comment:
enum feels a bit more natural here (avoiding 'stringly typed')

Reply:
and I'm also expecting (but have not checked) the possibility of a small performance improvements